lib/elliptic/ec/signature.js

```diff
@@ -23,35 +23,76 @@ function Signature(options, enc) {
 }
 module.exports = Signature;
 
+function getLength(buf, p) {
+  var initial = buf[p.place++];
+  if (!(initial & 0x80)) {
+    return initial;
+  }
+  var octetLen = initial & 0xf;
+  var byteLength = octetLen;
+  var val = buf[p.place + --byteLength];
```

Comment:
Any reason it can't be a loop like this?

```javascript
var val = 0;
for (var i = 0, off = p.place; i < byteLength; i++, off++) {
  val <<= 8;
  val |= buf[off];
}
```